src/deadline/houdini_adaptor/HoudiniAdaptor/adaptor.py

```diff
@@ -373,7 +374,19 @@ def _get_houdini_pathmap(self) -> str:
         path_mapping_rules: dict[str, str] = {}
 
         for rule in self._path_mapping_rules:
-            path_mapping_rules[rule.source_path] = rule.destination_path
+            source_path = rule.source_path
+            destination_path = rule.destination_path
+            rule_dict = rule.to_dict()
+
+            # Convert Windows paths to Posix as HOUDINI_PATHMAP turns \\ in
```

Comment:
I think instead of conditionally calling `as_posix()` when it's Windows, you can just call it regardless of OS. From the Python docs, it indicates that the function is a member of the parent `PurePath`. That would simplify the code to something like:

```python
# Houdini uses forward slashes on all OS
source_path = source_path.as_posix()
destination_path = destination_path.as_posix()
```

Reply:
I had tried that, but it doesn't seem to work Windows->Linux. I believe what's happening is when the rules are loaded on Linux it defaults to a PurePosixPath underneath which treats the `\` as valid path characters and not a separator to swap, so nothing is changed in the path.